src/io/ExportVTK.cpp

```diff
@@ -145,6 +146,81 @@ void ExportVTK::exportData(
   }
 }
 
+void ExportVTK::exportGradients(
+    std::ofstream &   outFile,
+    const mesh::Mesh &mesh)
+{
+  for (const mesh::PtrData &data : mesh.data()) {
+    if (data->hasGradient()) {
+      Eigen::MatrixXd &values = data->gradientValues();
+      if (data->getDimensions() > 1) {
+        const int dim = data->getDimensions();
+        int       colm{0};
+        const int datasize = values.cols() / dim;
+
+        outFile << "VECTORS " << data->getName() << "_dx"
+                << " double\n";
+        for (int j = 0; j < datasize; j++) { // Over columns
+          int i = 0;
+          for (; i < dim; i++) {
+            outFile << values.coeff(i, j) << ' ';
+          }
+          if (i < 3) {
+            outFile << '0';
```

Comment:
@DavidSCN here it should print a zero for third component

Reply:
Can you add your test case in the PR?